juju/juju
# feature/rework model defaults cli

Update tests accordingly.

Refs: juju-model-defaults-collapse

QA:
1. All unit tests pass
2. juju bootstrap reed-aws-west-1 awstest # with cloud like http://paste.ubuntu.com/23080517/
3. juju model-defaults # verify that the output matches what you'd expect from the config above.
4. juju model-defaults image-stream=develop test-mode=true # ensure settings are updated with juju model-config
5. juju model-defaults --reset image-stream test-mode # ensure they are unset
6. juju model-defaults image-stream
ATTRIBUTE     DEFAULT   CONTROLLER
image-stream  released  -
7. juju model-defaults --format=json image-stream
{"image-stream":{"default":"released"}}
8. Maybe try a few more


(Review request: http://reviews.vapour.ws/r/5616/)
